e2e/testsuite/testsuite.go

```diff
@@ -103,6 +141,111 @@ func (s *E2ETestSuite) configureGenesisDebugExport() {
+	ic := s.newInterchain(ctx, s.relayer, s.chains, channelOptionsModifier)
+
+	buildOpts := interchaintest.InterchainBuildOptions{
+		TestName:  s.T().Name(),
+		Client:    s.DockerClient,
+		NetworkID: s.network,
+		// we skip path creation because we are just creating the chains and not connections/channels
+		SkipPathCreation: true,
+	}
+
+	s.Require().NoError(ic.Build(ctx, s.GetRelayerExecReporter(), buildOpts))
+}
+
+// SetupTest will by default use the default channel options to create a path between the chains.
+// if non default channel options are required, the test suite must override the `SetupTest` function.
+func (s *E2ETestSuite) SetupTest() {
+	s.SetupPath(ibc.DefaultClientOpts(), defaultChannelOpts(s.GetAllChains()))
+}
+
+// SetupPath creates a path between the chains using the provided client and channel options.
+func (s *E2ETestSuite) SetupPath(clientOpts ibc.CreateClientOptions, channelOpts ibc.CreateChannelOptions) {
+	s.T().Logf("Setting up path for: %s", s.T().Name())
+	r := s.relayer
+
+	ctx := context.TODO()
+	allChains := s.GetAllChains()
+	for i := 0; i < len(allChains)-1; i++ {
+		chainA, chainB := allChains[i], allChains[i+1]
+		pathName := s.generatePathName()
+		s.T().Logf("establishing path between %s and %s on path %s", chainA.Config().ChainID, chainB.Config().ChainID, pathName)
+
+		err := r.GeneratePath(ctx, s.GetRelayerExecReporter(), chainA.Config().ChainID, chainB.Config().ChainID, pathName)
+		s.Require().NoError(err)
+
+		// Create new clients
+		err = r.CreateClients(ctx, s.GetRelayerExecReporter(), pathName, clientOpts)
+		s.Require().NoError(err)
+		err = test.WaitForBlocks(ctx, 1, chainA, chainB)
+		s.Require().NoError(err)
+
+		err = r.CreateConnections(ctx, s.GetRelayerExecReporter(), pathName)
+		s.Require().NoError(err)
+		err = test.WaitForBlocks(ctx, 1, chainA, chainB)
+		s.Require().NoError(err)
+
+		err = r.CreateChannel(ctx, s.GetRelayerExecReporter(), pathName, channelOpts)
+		s.Require().NoError(err)
+		err = test.WaitForBlocks(ctx, 1, chainA, chainB)
+		s.Require().NoError(err)
+
+		channelsA, err := r.GetChannels(ctx, s.GetRelayerExecReporter(), chainA.Config().ChainID)
+		s.Require().NoError(err)
+
+		channelsB, err := r.GetChannels(ctx, s.GetRelayerExecReporter(), chainB.Config().ChainID)
+		s.Require().NoError(err)
+
+		if s.channels[s.T().Name()] == nil {
+			s.channels[s.T().Name()] = make(map[ibc.Chain][]ibc.ChannelOutput)
+		}
+
+		// keep track of channels associated with a given chain for access within the tests.
+		s.channels[s.T().Name()][chainA] = channelsA
+		s.channels[s.T().Name()][chainB] = channelsB
+		s.testPaths[s.T().Name()] = append(s.testPaths[s.T().Name()], pathName)
+	}
+}
+
+// GetChannel returns the ibc.ChannelOutput for the current test.
+// this defaults to the first entry in the list, and will be what is needed in the case of
+// a single channel test.
+func (s *E2ETestSuite) GetChannel() ibc.ChannelOutput {
```

Comment:
Would it better to rename this to be more precise? Either something like GetChainAChannel or refactor it to take in chain1 and chain2 to get the channel between them? A bit more verbose, but also potentially useful in more contexts?

I guess my lens is a bit skewed with all the forwarding tests, so feel free to ignore :D

Reply:
Not a bad idea to be explicit here, I think `GetChainAChannel` would be better.